### PR TITLE
fixed initialization of reset and interrupt pins (BSP-321)

### PR DESCRIPTION
**Pull request by ivanroberto**
# ESP-BSP Pull Request checklist

- [master] Version of modified component bumped

# Change description
There is an error in initializing the reset and interrupt pins for the FT5X06 touch driver, where part of the pins' functionalities were changed.

### CODE REVIEW

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c [hunk 1]
```diff
@@ -120,28 +120,28 @@ esp_err_t esp_lcd_touch_new_i2c_ft5x06(const esp_lcd_panel_io_handle_t io, const
 
     /* Prepare pin for touch interrupt */
     if (esp_lcd_touch_ft5x06->config.int_gpio_num != GPIO_NUM_NC) {
-        const gpio_config_t int_gpio_config = {
+        const gpio_config_t rst_gpio_config = {
```
Comment (espzav) [inline components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:123]: The name of variable was right.
```suggestion
        const gpio_config_t int_gpio_config = {
```

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c [hunk 2]
```diff
@@ -120,28 +120,28 @@ esp_err_t esp_lcd_touch_new_i2c_ft5x06(const esp_lcd_panel_io_handle_t io, const
...
             .mode = GPIO_MODE_INPUT,
+            .intr_type = GPIO_INTR_NEGEDGE,
             .pin_bit_mask = BIT64(esp_lcd_touch_ft5x06->config.int_gpio_num)
         };
-        ret = gpio_config(&int_gpio_config);
+        ret = gpio_config(&rst_gpio_config);
```
Comment (espzav) [inline components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:128]: Same here.

```suggestion
        ret = gpio_config(&int_gpio_config);
```

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c [hunk 3]
```diff
@@ -120,28 +120,28 @@ esp_err_t esp_lcd_touch_new_i2c_ft5x06(const esp_lcd_panel_io_handle_t io, const
...
+        /* Register interrupt callback */
+        if (esp_lcd_touch_ft5x06->config.interrupt_callback) {
+            esp_lcd_touch_register_interrupt_callback(esp_lcd_touch_ft5x06, esp_lcd_touch_ft5x06->config.interrupt_callback);
+        }
     }
 
     /* Prepare pin for touch controller reset */
     if (esp_lcd_touch_ft5x06->config.rst_gpio_num != GPIO_NUM_NC) {
-        const gpio_config_t rst_gpio_config = {
+        const gpio_config_t int_gpio_config = {
```
Comment (espzav) [inline components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:139]: The name of variable was right.
```suggestion
        const gpio_config_t rst_gpio_config = {
```

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c [hunk 4]
```diff
@@ -120,28 +120,28 @@ esp_err_t esp_lcd_touch_new_i2c_ft5x06(const esp_lcd_panel_io_handle_t io, const
...
             .mode = GPIO_MODE_OUTPUT,
-            .intr_type = GPIO_INTR_NEGEDGE,
             .pin_bit_mask = BIT64(esp_lcd_touch_ft5x06->config.rst_gpio_num)
         };
-        ret = gpio_config(&rst_gpio_config);
+        ret = gpio_config(&int_gpio_config);
```
Comment (espzav) [inline components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:143]:
```suggestion
        ret = gpio_config(&rst_gpio_config);
```

**Review by espzav — COMMENTED** [review]
Thank you for this fix! It is my copy-paste mistake. Please, rename the variables, otherwise LGTM.
Edit: Please, increase the version of the component.

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c (hunk 1 above)
Comment (espzav) [reply components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c]: This change wasn't fixed.

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c (hunk 2 above)
Comment (espzav) [reply components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:128]: This change wasn't fixed.

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c (hunk 4 above)
Comment (espzav) [reply components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:143]: This change wasn't fixed.

**Comment by tore-espressif** [comment]
@espzav thank you for the review.

@ivanroberto do you plan to resolve the issues, or should we take over the PR?

File: components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c (hunk 4 above)
Comment (ivanroberto) [reply components/lcd_touch/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c]: I already just push again with the correct names, sorry

**Review by tore-espressif — APPROVED** [review]
LGTM!

@espzav PTAL too. We can squash to commits and update the version ourselves, after this is merged

**Pull request merged**